course_catalog/serializers.py

```diff
@@ -568,6 +569,18 @@ def validate_list_type(self, list_type):
             raise ValidationError("Missing/incorrect list type information")
         return list_type
 
+    def validate_privacy_level(self, privacy_level):
+        """
+        Validator for privacy_level
+        """
+        request = self.context.get("request")
+        if request and hasattr(request, "user") and isinstance(request.user, User):
+            if privacy_level == PrivacyLevel.public.value and not is_list_staff(
+                request.user
+            ):
+                raise ValidationError("Invalid permissions for creating public lists")
```

Comment:
This error could be hit when updating a user list, could you make the error message more general to both cases?